components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/jwks/JwksEndpoint.java

```diff
@@ -72,21 +72,18 @@ public String jwks() {
 
         String tenantDomain = getTenantDomain();
         String keystorePath = CarbonUtils.getServerConfiguration().getFirstProperty(SECURITY_KEY_STORE_LOCATION);
```

Comment:
Another small thing I noticed was that, 

Loading the keystore form `keystorePath` is only required for super.tenant.

But we are opening a file input stream regardless of whether its super tenant or not

Reply:
Fixed :)

Reply:
If we move the `FileInputStream` away from the try-with pattern and into the "if" block of the super tenant, we will have to close the resource inside the try itself. Is this alright?